File: sem4/mpp/ex2/repository/src/main/java/repository/Repository.java
package repository;

import domain.Identifiable;

import java.io.Serializable;
import java.util.Collection;

public interface Repository<T extends Identifiable<ID>, ID extends Serializable> {
	T add(T elem);
	void delete(ID id);
	void update(T elem, ID id);
	T getById(ID id);
	Collection<T> getAll();
}

File: sem4/mpp/project-cs/repository/Repository.cs
using project_cs.domain;

namespace project_cs.repo;

public interface IRepository<T, TId> where T : Identifiable<TId> {
    T Add(T elem);
    void Delete(TId id);
    void Update(T elem, TId id);
    T? GetById(TId id);
    ICollection<T> GetAll();
}